internal/provider/vapp/isolated_network_resource.go

```diff
@@ -0,0 +1,511 @@
+		return
+	}
+
+	// Get UUID.
+	networkID, err := govcd.GetUuidFromHref(vAppNetwork.Link.HREF, false)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating vApp network uuid", err.Error())
+		return
+	}
+
+	id := common.NormalizeID("urn:vcloud:network:", networkID)
+
+	plan.ID = types.StringValue(id)
+
+	// Set state to fully populated data
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *isolatedNetworkResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var (
+		state *isolatedNetworkResourceModel
+		diag  diag.Diagnostics
+	)
+
+	// Get current state
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	isolatedNetworkRef, errInit := state.initNetworkQuery(ctx, r.client, false)
+	if errInit != nil {
+		if errInit.Summary == ErrVAppNotFound {
+			resp.State.RemoveResource(ctx)
+		}
+		resp.Diagnostics.AddError(errInit.Summary, errInit.Detail)
+		return
+	}
+
+	if isolatedNetworkRef.VAppLocked {
+		defer isolatedNetworkRef.VAppUnlockF()
+	}
+
+	vAppNetworkConfig, err := isolatedNetworkRef.VApp.GetNetworkConfig()
+	if err != nil {
+		resp.Diagnostics.AddError("error getting vApp networks", err.Error())
+		return
+	}
+
+	vAppNetwork := govcdtypes.VAppNetworkConfiguration{}
+	for _, networkConfig := range vAppNetworkConfig.NetworkConfig {
+		if networkConfig.NetworkName == state.Name.ValueString() {
+			vAppNetwork = networkConfig
+		}
+	}
+
+	if vAppNetwork == (govcdtypes.VAppNetworkConfiguration{}) {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	// Get UUID.
+	networkID, err := govcd.GetUuidFromHref(vAppNetwork.Link.HREF, false)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating vApp network uuid", err.Error())
+		return
+	}
+
+	id := common.NormalizeID("urn:vcloud:network:", networkID)
+
+	plan := &isolatedNetworkResourceModel{
+		ID:                 types.StringValue(id),
+		VDC:                state.VDC,
+		Name:               types.StringValue(vAppNetwork.NetworkName),
+		Description:        types.StringValue(vAppNetwork.Description),
+		VAppName:           state.VAppName,
+		Netmask:            types.StringValue(vAppNetwork.Configuration.IPScopes.IPScope[0].Netmask),
+		Gateway:            types.StringValue(vAppNetwork.Configuration.IPScopes.IPScope[0].Gateway),
+		DNS1:               types.StringValue(vAppNetwork.Configuration.IPScopes.IPScope[0].DNS1),
+		DNS2:               types.StringValue(vAppNetwork.Configuration.IPScopes.IPScope[0].DNS2),
+		DNSSuffix:          types.StringValue(vAppNetwork.Configuration.IPScopes.IPScope[0].DNSSuffix),
+		GuestVLANAllowed:   types.BoolValue(*vAppNetwork.Configuration.GuestVlanAllowed),
+		RetainIPMacEnabled: types.BoolValue(*vAppNetwork.Configuration.RetainNetInfoAcrossDeployments),
+	}
```

Comment:
For preventing `nil` pointer check if `vAppNetwork.Configuration`, `vAppNetwork.Configuration.IPScopes` is not nil. 

Check if array `vAppNetwork.Configuration.IPScopes.IPScope` a least 1

Reply:
As discussed in Teams, it's not a best practice to test all pointers return by a previous function.
I just add the length test for IPScope array.